android/modules/calendar/src/java/ti/modules/titanium/calendar/CalendarProxy.java

```diff
@@ -188,12 +198,147 @@ public EventProxy getEventById(int id)
+	}
+
 	@Kroll.method
 	public EventProxy createEvent(KrollDict data)
 	{
 		return EventProxy.createEvent(this, data);
 	}
 
+	@Kroll.method
+	public EventProxy[] createEvents(Object data)
+	{
+		// Validate arguments to be an array.
+		if (!(data instanceof Object[] dataList && dataList.length > 0)) {
+			Log.e(TAG, "Argument expected to be an array.");
+			return null;
+		}
+
+		// Check for permissions.
+		ContentResolver contentResolver = TiApplication.getInstance().getContentResolver();
+		if (!CalendarProxy.hasCalendarPermissions()) {
+			Log.e(TAG, "Calendar permissions are missing.");
+			return null;
+		}
+
+		ArrayList<ContentProviderOperation> operations = new ArrayList<>();
+		ArrayList<EventProxy> eventProxies = new ArrayList<>();
+		Map<Integer, Integer> proxyResultIndexMapping = new HashMap<>();
+
+		for (int i = 0, firstIndex = 0; i < dataList.length; i++) {
+			KrollDict krollDict = new KrollDict((HashMap) dataList[i]);
+
+			EventProxy eventProxy = new EventProxy();
+			ContentValues contentValues = CalendarUtils.createContentValues(this, krollDict, eventProxy);
+
+			// We cannot pass null data to ContentProviderOperation.
+			// Necessary to keep track of non-null items later.
+			if (contentValues == null) {
+				eventProxies.add(null);
+				Log.e(TAG, "Title was not created, no title found for event");
+				continue;
+			}
+
+			ContentProviderOperation.Builder builder = ContentProviderOperation
+				.newInsert(CalendarContract.Events.CONTENT_URI)
+				.withValues(contentValues);
+
+			operations.add(builder.build());
+			eventProxies.add(eventProxy);
+
+			proxyResultIndexMapping.put(i, firstIndex);
+			firstIndex++;
+		}
+
+		try {
+			// Execute the batch operation
+			ContentProviderResult[] results = contentResolver.applyBatch(
+				getEventsUri().getAuthority(),
+				operations
+			);
+
+			// Find non-null proxies and map their IDs
+			for (int proxyIndex : proxyResultIndexMapping.keySet()) {
+				int proxyIndexInResults = proxyResultIndexMapping.get(proxyIndex);
+				Uri eventUri = results[proxyIndexInResults].uri;
+
+				if (eventUri != null) {
+					// Set event id to proxy.
+					eventProxies.get(proxyIndex).id = eventUri.getLastPathSegment();
+				} else {
+					// Event failed to get a proper URI path, should set to null in this case too.
+					eventProxies.set(proxyIndex, null);
+				}
+			}
+
+			return eventProxies.toArray(new EventProxy[0]);
+
+		} catch (RemoteException | OperationApplicationException e) {
+			Log.e(TAG, "Batch insert operation failed: " + e.getMessage());
+			return null;
+		}
+	}
+
+	@Kroll.method
+	public int deleteEvents(Object args)
+	{
+		int deletedCount = 0;
+
+		// Validate arguments to be an array.
+		if (!(args instanceof Object[] eventIds && eventIds.length > 0)) {
+			Log.e(TAG, "Argument expected to be an array.");
+			return deletedCount;
+		}
+
+		// Check for permissions.
+		ContentResolver contentResolver = TiApplication.getInstance().getContentResolver();
```

Comment:
move that into the `try` part so it is closer the the place where it is used and if we don't have permissions we don't need to get it.